From 8ed4a97f0abf6406803fa765240cfd5addcc6fd7 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:29 -0400
Subject: [tools] perf/report: Print session information only if --stdio is given

Message-id: <1408953210-26343-202-git-send-email-jolsa@redhat.com>
Patchwork-id: 88977
O-Subject: [PATCH RHEL7.1 BZ1133083 201/322] perf report: Print session information only if --stdio is given
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 150e465ac99ed18fb9555c16e0def7ce01913a2a
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Fri Dec 20 14:11:13 2013 +0900

description
===========
Move those print functions under "if (use_browser == 0)" so that they
don't interfere with TUI output.

Maybe they can handle other UIs later.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index 0c9ec3e..bf8dd2e 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -469,15 +469,17 @@ static int __cmd_report(struct report *rep)
   desc);
  }
 
- if (verbose > 3)
-  perf_session__fprintf(session, stdout);
+ if (use_browser == 0) {
+  if (verbose > 3)
+   perf_session__fprintf(session, stdout);
 
- if (verbose > 2)
-  perf_session__fprintf_dsos(session, stdout);
+  if (verbose > 2)
+   perf_session__fprintf_dsos(session, stdout);
 
- if (dump_trace) {
-  perf_session__fprintf_nr_events(session, stdout);
-  return 0;
+  if (dump_trace) {
+   perf_session__fprintf_nr_events(session, stdout);
+   return 0;
+  }
  }
 
  nr_samples = 0;
-- 
1.7.1